content/setup/local-with-vagrant/_index.md

```diff
@@ -0,0 +1,198 @@
+here.
+
+SSH in the provisioner again and we have to define our hello world template and load it to Tinkerbell:
+
+```
+$ cat > hello-world.yml
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "{{.device_1}}"
+    actions:
+      - name: "hello_world"
+        image: hello-world
+        timeout: 60
+$ docker exec -i deploy_tink-cli_1 tink template create --name hello-world < ./hello-world.yml
+Created Template:  75ab8483-6f42-42a9-a80d-a9f6196130df
+```
+
+Register the worker with Tinkerbell
+
+```
+$ cat > hardware-data.json
+{
+  "id": "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+  "arch": "x86_64",
+  "allow_pxe": true,
+  "allow_workflow": true,
+  "facility_code": "onprem",
+  "ip_addresses": [
+    {
+      "address": "192.168.1.5",
+      "address_family": 4,
+      "enabled": true,
+      "gateway": "192.168.1.1",
+      "management": true,
+      "netmask": "255.255.255.248",
+      "public": false
+    }
+  ],
+  "network_ports": [
+    {
+      "data": {
+        "mac": "08:00:27:00:00:01"
+      },
+      "name": "eth0",
+      "type": "data"
+    }
+  ]
+}
+
+$ docker exec -i deploy_tink-cli_1 tink hardware push < ./hardware-data.json
+2020/06/17 14:12:45 Hardware data pushed successfully
+```
+
+Now that we have a template and hardware registered we can trigger the workflow for that particular hardware but first a bit of context on what we just did:
+
+1. A template is used to create workflows. Template are important because you
+   can create as many workflows you need from a single template.
+2. A workflow needs a target (a hardware) in vagrant it is the worker and it is
+   identified via MAC address.
+
+Let's start the workflow where `-t` is the template ID you got when registering
+your template and `-r` contains the variables to replace in the template, in
+this case only the worker's MAC address is required:
+
+```
+$ docker exec -i deploy_tink-cli_1 tink workflow create \
+    -t 75ab8483-6f42-42a9-a80d-a9f6196130df \
+    -r '{"device_1":"08:00:27:00:00:01"}'
+Created Workflow:  a8984b09-566d-47ba-b6c5-fbe482d8ad7f
+```
+
+> Note: this MAC address it hard coded in the Vagrantfile.
+
+We need to spin up our first worker now.
+
+## Start a worker
+
+Open a new terminal and get inside the `tink/deploy/vagrant`.
+
+For a worker the MAC address works as a unique identifier, with Vagrant we are
+able to set it to fixed value, in this case `080027000001`.
+
+As you did for the provisioner start the worker via vagrant:
+
+```
+$ vagrant up worker
+```
+
+If using VirtualBox, the worker shows up in a UI and if everything is right you
+should see at the end of the boot a login to a custom Alpine that Tinkerbell
+netboot. The operating system runs on RAM, the changes won't be persisted across
+reboots. You can login with the username `root` and no password is required.
+
+![Screenshot from the worker](/images/vagrant-setup-vbox-worker.png)
```

Comment:
Perhaps show the user how to enlarge the size of the console? For MacOS users with Retina or 4k screens, it appears as an unreadable size:

<img width="472" alt="Screenshot 2020-06-18 at 12 04 25" src="https://user-images.githubusercontent.com/6358735/85013922-496a4e80-b15d-11ea-8d70-52a26d5f98b3.png">

<img width="1192" alt="Screenshot 2020-06-18 at 12 04 40" src="https://user-images.githubusercontent.com/6358735/85013912-45d6c780-b15d-11ea-98ff-3d8a074d9f04.png">

<img width="763" alt="Screenshot 2020-06-18 at 12 04 33" src="https://user-images.githubusercontent.com/6358735/85013920-48d1b800-b15d-11ea-95f6-789ee1f85c8e.png">